components/compositing/webview.rs

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::HashMap;
+
+use msg::constellation_msg::TopLevelBrowsingContextId;
+
+#[derive(Debug)]
+pub struct WebViewManager<WebView> {
```

Comment:
Why is this generic on WebView? Should there only be a single implementation of WebView?

Reply:
since WebViewManager only uses the type for opaque storage, the unit tests can store dummy values that are cloneable and debuggable and so on, instead of all the data we actually store against each webview, making the tests easier to write and understand. same goes for WebViewManager in the constellation.